client/my-sites/plans/jetpack-plans/get-purchase-url-callback.ts

```diff
@@ -104,8 +104,10 @@ export const buildUpsellURL = (
 
 	// If upsell exists
 	if ( product in PURCHASE_FLOW_UPSELLS_MATRIX ) {
-		if ( ! urlQueryArgs.checkoutBackUrl ) {
+		if ( isJetpackCloud() && ! urlQueryArgs.checkoutBackUrl ) {
 			urlQueryArgs.checkoutBackUrl = window.location.href;
+		} else if ( ! urlQueryArgs.cancel_to ) {
```

Comment:
Do I understand correctly here, that if the env is Jetpack Cloud, and checkoutBackUrl is already provided, we want to set the `cancel_to` property?

Or should the `cancel_to` property only be set on Calypso Blue?

Reply:
`checkoutBackUrl` is only useful in the Cloud context. That's why I added the additional check.